# FAVA: Functional Associations using VAriational Autoencoders

![Picture 1](https://user-images.githubusercontent.com/81096946/172628567-2b8f2bbc-bec5-4859-87a4-3bc7c0e72099.png)

Protein networks are commonly used for understanding the interplay between proteins in the cell as well as for visualizing omics data. Unfortunately, existing networks such as STRING are heavily biassed by data availability in the sense that well-studied proteins have many more interactions than understudied proteins. To create networks also for the latter, we need to use high-throughput data, such as single cell RNA-seq (scRNA-seq) and proteomics, which do not have this literature bias. However, due to the sparseness (i.e. many proteins not observed in each cell/sample) and redundancy (many similar cells/samples) of such data, simple correlation analysis does not result in high-quality networks. We present FAVA, Functional Associations using Variational Autoencoders, which deals with these issues by compressing the high-dimensional data into a meaningful, dense, low-dimensional latent space. We demonstrate that calculating correlations in this latent space results in much improved networks compared to the original representation for enormous scRNA-seq and proteomics data from Human Protein Atlas and PRIDE, respectively. We show that these networks, which given the nature of the input data should be free of literature bias, indeed have much better coverage of understudied proteins than existing networks.
